### PR TITLE
Add examples for refspecs

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/libgit2/remote.jl [hunk 1]
```diff
@@ -186,6 +203,19 @@ end
...
+
+julia> LibGit2.add_push!(repo, remote, "refs/heads/master");
+
+julia> close(remote);
+
+julia> remote = LibGit2.get(LibGit2.GitRemote, repo, "upstream");
+
+julia> LibGit2.push_refspecs(remote)
+String["refs/heads/master"]
 """
```
Comment (tkelman) [inline base/libgit2/remote.jl]: missing closing backtick

File: base/libgit2/remote.jl [hunk 2]
```diff
@@ -68,7 +68,14 @@ end
 
 Determine if the `remote_name` specified exists within the `repo`. Returns a
 [`Nullable`](@ref), which will be null if the requested remote does not exist. If the remote
-does exist, the `Nullable` contains a `GitRemote` to the remote name.
+does exist, the `Nullable` contains a [`GitRemote`](@ref) to the remote name.
+
+# Examples
+```julia-repl
```
Comment (tkelman) [inline base/libgit2/remote.jl]: just julia

File: base/libgit2/remote.jl [hunk 3]
```diff
@@ -205,7 +205,7 @@ Get the *push* refspecs for the specified `rmt`. These refspecs contain
 information about which branch(es) to push to.
 
 # Examples
-```julia-repl
+```julia
```
Comment (tkelman) [inline base/libgit2/remote.jl]: this one is julia-repl since it has the prompt

Comment (kshyatt) [inline base/libgit2/remote.jl]: aaaaaaaaa self-owned yet again!

Comment (tkelman) [inline base/libgit2/remote.jl]: :coffee:

**Review by Sacha0 — APPROVED** [review]
superficially lgtm! :)

File: base/libgit2/remote.jl (hunk 3 above)
Comment (kshyatt) [inline base/libgit2/remote.jl]: The song "16 tons", but for ounces of coffee...

**Review by ararslan — APPROVED** [review]

**Comment by kshyatt** [comment]
@tkelman ok to squash?

**Review by tkelman — APPROVED** [review]

**Pull request merged**